shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -110,22 +114,33 @@ class ExpandedEntity private constructor(
         }
     }
 
-    fun checkPropertiesHaveAtMostOneDefaultInstance() {
-        properties.forEach { property ->
-            if (property.value.count { !it.containsKey(NGSILD_DATASET_ID_PROPERTY) } > 1)
-                throw BadRequestDataException("Property ${property.key.extractShortTypeFromExpanded()} can't have more than one default instance")
+    fun checkAttributesHaveAtMostOneDefaultInstance() {
+        properties.checkAttributesHaveAtMostOneDefaultInstance()?.let { throw BadRequestDataException("Property ${it.extractShortTypeFromExpanded()} can't have more than one default instance") }
+        relationships.checkAttributesHaveAtMostOneDefaultInstance()?.let { throw BadRequestDataException("Relationship ${it.extractShortTypeFromExpanded()} can't have more than one default instance") }
+    }
+
+    fun checkAttributesHaveUniqueDatasetId() {
+        properties.checkAttributesHaveUniqueDatasetId()?.let { throw BadRequestDataException("Property ${it.extractShortTypeFromExpanded()} can't have duplicated datasetId") }
+        relationships.checkAttributesHaveUniqueDatasetId()?.let { throw BadRequestDataException("Relationship ${it.extractShortTypeFromExpanded()} can't have duplicated datasetId") }
+    }
+
+    private fun Map<String, List<Map<String, List<Any>>>>.checkAttributesHaveAtMostOneDefaultInstance(): String? {
```

Comment:
Not really fan of the method contract.

Reply:
We have to guess that a method called `checkSomething` returns `null` if the check has failed

Reply:
findAttributeWithMoreThanOneDefaultInstance ?

Reply:
findAttributeWithMoreThanOneDefaultInstance is more meaningful.

Reply:
go for this renaming and close this PR.